aion/lib/aion/room_channel/room.ex

```diff
@@ -76,12 +89,24 @@ defmodule Aion.RoomChannel.Room do
     room.question
   end
 
-  @spec get_new_question_with_answers(integer) :: %{question: Question.t, answers: list Answer.t}
-  defp get_new_question_with_answers(room_id) do
-      question = Question.get_random_question(room_id)
-      answers = Answer.get_answers(question.id)
-      Logger.debug fn -> "Answers: #{inspect(Enum.map(answers, fn answer -> answer.content end))}" end
+  @spec get_new_question_with_answers(question_list, integer) ::  new_question_set
+  defp get_new_question_with_answers(questions, room_id) do
+    [new_question | remaining_questions] = questions
```

Comment:
This match does not seem to be safe. You are assuming there will always be at least one question in the category. What happens if user creates three categories and adds no questions to any of them? A more intuitive approach would be to let this function be called with an empty list of questions. Here you would do a pattern match like this:
```
case questions do
[next_question | remaining_questions] -> you know there's a question you can display now
[] -> okay, the questions are finished (or there were none in the first place) and you can trigger end of the game from here
```